### PR TITLE
plugin/nvim-lightbulb: update module to match latest breaking changes

### LINKED ISSUE #843: nvim-lightbulb plugin requires updating to support latest features
**Plugin affected**: nvim-lightbulb
**Nixpkgs channel**: unstable
**Home-manager version**: latest master

## Description
nvim-lightbulb reached v1.0.0 back in July ([02aacd9](https://github.com/kosayoda/nvim-lightbulb/commit/02aacd93b9b2367e6dbdd20f43b525f701faf834)) which introduced new some new handlers (number, line) and possibly broke some users configurations.

Seemed like a simple update to get these new features added, and to restructure the default config to match the one shown on the projects [README.md](https://github.com/kosayoda/nvim-lightbulb/blob/master/README.md). So, I will shortly follow up with a pull request.

## Config
```nix
nvim-lightbulb = {
  enable = true;
  autocmd.enabled = true;
  # virtualText.enabled = true;
  number.enabled = true; # does nothing
  sign.enabled = false;
};
```

### PR DESCRIPTION
**Pull request by rsHalford**
Closes #843.

Could someone please double check line 40 (actionKinds)? As I'm not exactly sure how to get to a `nil` value in lua from nix.

### CODE REVIEW

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
From a first glance, this looks nice.
If some options are removed/renamed in the process, please use `mkRemovedOptionModule` and `mkRenamedOptionModule` to properly warn the users.
You will be able to find some examples using those in the repo. ([here](https://github.com/nix-community/nixvim/blob/310f1ad5b5e7da20832b216a64f6da528331db4a/plugins/filetrees/neo-tree.nix#L12-L23) for instance).

**Comment by rsHalford** [comment]
Is there a suggested way to create a notice for `plugins.nvim-lightbulb.ignore` (list of strings) now being `plugins.nvim-lightbulb.ignore.clients`?
`mkRenamedOptionModule` doesn't seem like a good fit and I can't seem to find an alternative within [modules.nix](https://github.com/NixOS/nixpkgs/blob/master/lib/modules.nix).

(I'm most likely wrong and just haven't seen it)

**Comment by GaetanLepage** [comment]
> Is there a suggested way to create a notice for `plugins.nvim-lightbulb.ignore` (list of strings) now being `plugins.nvim-lightbulb.ignore.clients`? `mkRenamedOptionModule` doesn't seem like a good fit and I can't seem to find an alternative within [modules.nix](https://github.com/NixOS/nixpkgs/blob/master/lib/modules.nix).
> 
> (I'm most likely wrong and just haven't seen it)

You are right, in this case, there is not much we can do.
Let's just not add a warning for this one.

File: plugins/lsp/nvim-lightbulb.nix [hunk 1]
```diff
@@ -6,78 +6,178 @@
   ...
 }:
 with lib; {
+  imports = [
+    (
+      mkRemovedOptionModule
+      ["plugins" "nvim-lightbulb" "sign" "priority"]
+      "Use `plugins.nvim-lightbulb.priority` to set lightbulb priority for all handlers except float."
+    )
+  ];
```
Comment (GaetanLepage) [inline plugins/lsp/nvim-lightbulb.nix:15]:
```suggestion
  imports = let
    basePluginPath = ["plugins" "nvim-lightbulb"];
  in [
    (
      mkRenamedOptionModule
      (basePluginPath ++ ["sign" "priority"])
      (basePluginPath ++ ["priority"])
    )
  ];
```

Comment (rsHalford) [reply plugins/lsp/nvim-lightbulb.nix]: Thanks for pointing that out for me!

**Review by GaetanLepage — APPROVED** [review]

**Review by traxys — APPROVED** [review]

**Pull request merged**